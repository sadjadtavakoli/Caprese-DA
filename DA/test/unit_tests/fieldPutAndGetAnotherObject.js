let obj1 = {
    name:"sadjad",
    job: "researcher",
    talk: function(){
        let variable = this.name
    }
}

let obj2 = {
    name:"berke",
    job: "student",
    talk: function(){
        obj1.name = "palviz"
        obj1.talk()
    }
}

obj2.talk()
let var1 = obj1.name

// obj2.talk impacts obj1.talk
// obj2.talk impacts main
