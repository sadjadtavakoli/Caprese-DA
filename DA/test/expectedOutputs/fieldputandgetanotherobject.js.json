{"talk-test/unit_tests/fieldputandgetanotherobject.js-12-15":{"impacted":["t_1","t_11"]},"talk-test/unit_tests/fieldputandgetanotherobject.js-4-6":{"impacted":["t_4"]},"keyMap":{"t_1":"test/unit_tests/fieldputandgetanotherobject.js","t_4":"talk-test/unit_tests/fieldputandgetanotherobject.js-12-15","t_11":"talk-test/unit_tests/fieldputandgetanotherobject.js-4-6"}}